lib/bioschemas/DataRecord.js

```diff
@@ -16,7 +16,7 @@ module.exports = function generateDataRecord(metadata) {
     "mainEntity": {
       "@type": metadata.rdfType,
       "identifier": metadata.name,
-      "hasBioChemRole": metadata.roles.map((role) => role.uri),
+      "hasBioChemRole": metadata.roles?metadata.roles.map((role) => role.uri):[],
```

Comment:
This operation is too complex for an in-line ternary operation, it is hard to read

Reply:
I think I prefer to keep this one.